src/main/scala/tech/cryptonomic/conseil/tezos/TezosNodeOperator.scala

```diff
@@ -179,33 +177,36 @@ class TezosNodeOperator(val node: TezosRPCInterface, batchConf: BatchFetchConfig
         } else {
           if (maxLevel == -1) logger.warn("There were apparently no blocks in the database. Downloading the whole chain..")
           else logger.info("Found new block head at level {}, currently stored max is {}. Fetching missing blocks", headLevel, maxLevel)
-          getBlocks(network, headHash, maxLevel + 1, headLevel, followFork)
+          getBlocks(network, headHash, maxLevel + 1, headLevel)
         }
     } yield blocks
 
   /**
-    * Get the blocks in a specified range.
-    * @param network Which tezos network to go against
-    * @param minLevel Minimum block level
-    * @param maxLevel Maximum block level
-    * @param startBlockHash If specified, start from the supplied block hash, otherwise, head of the chain.
-    * @param followFork If predecessor of the minLevel block appears to be on a fork, also capture the blocks on the fork.
-    * @return Blocks
+    * Gets all blocks.
+    * @param network    Which Tezos network to go against
+    * @return           Blocks and Account hashes involved
     */
-  def getBlocks(network: String,
-                startBlockHash: BlockHash = blockHeadHash,
-                minLevel: Int,
-                maxLevel: Int,
-                followFork: Boolean): Future[List[(Block, List[AccountId])]] = {
-    val blocksInRange = processBlocks(network, startBlockHash, minLevel, maxLevel)
-    val blocksFromFork =
-      if (followFork) Future.successful(List.empty)
-      else Future.successful(List.empty)
+  def getAllBlocks(network: String): Future[List[(Block, List[AccountId])]] =
     for {
-      forkBlocks <- blocksFromFork
-      rangeBlocks <- blocksInRange
-    } yield forkBlocks ++ rangeBlocks
-  }
+      blockHead <- getBlockHead(network)
+      headLevel = blockHead.metadata.header.level
+      headHash = blockHead.metadata.hash
+      blocks <- getBlocks(network, headHash, 1, headLevel)
+    } yield blocks
+
+  /**
+    * Gets last n blocks.
+    * @param network    Which Tezos network to go against
+    * @param amount     Amount of latest block to fetch
+    * @return           Blocks and Account hashes involved
+    */
+  def getLatestBlocks(network: String, amount: Int): Future[List[(Block, List[AccountId])]] =
+    for {
+      blockHead <- getBlockHead(network)
```

Comment:
Can we reduce this to a single call for the head and the previous `amount`?

Reply:
I'm not sure what do you mean. I renamed ``amount`` to ``count`` because I suppose that it was misleading :)

Reply:
Ok, now I understand. It's probably possible but it affects many other methods as well. I've created a task to investigate this: https://github.com/Cryptonomic/Conseil/issues/229

Reply:
It was not a mandatory requirement, I was just asking to know if it could be simplified *as is**.

I didn't meant to imply further work to make it possible, so it's ok like this